xrpl/models/transactions/credential_accept.py

```diff
@@ -0,0 +1,63 @@
+"""Model for CredentialAccept transaction type."""
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+from typing_extensions import Self
+
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import HEX_REGEX, KW_ONLY_DATACLASS, require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class CredentialAccept(Transaction):
+    """This transaction accepts a credential issued to the Account (i.e. the Account is
+    the Subject of the Credential object). The credential is not considered valid until
+    it has been transferred/accepted.
+    """
+
+    transaction_type: TransactionType = field(
+        default=TransactionType.CREDENTIAL_ACCEPT,
+        init=False,
+    )
+
+    account: str = REQUIRED  # type: ignore
+    """
+    The subject of the credential.
+    """
+    issuer: str = REQUIRED  # type: ignore
+    """
+    The issuer of the credential.
+    """
+
+    credential_type: str = REQUIRED  # type: ignore
+    """
+    A (hex-encoded) value to identify the type of credential from the issuer.
+    """
+
+    # Note: Validity checks pertaining to other fields of this transaction require
+    # access to the latest state of the blockchain
```

Comment:
This is unneeeded. The library doesn't need to validate every single piece of a transaction, that's what rippled is doing. The error checking is just to do some additional sanity checking/for the purpose of failing fast.

Reply:
I will remove the comment.

I find this argument tenuous. Any of these violations would have resulted in a `tem...` error code. Users would not face any financial slap-on-the-wrist for these violations, its only a matter of few seconds.

I don't like this haphazard duplication of effort across rippled and the client libraries. For instance, requests do not have any of these validity checks.

Reply:
fixed in ad4331a0bab1f4d603f07fc3d49c1a52b2b0938d

Reply:
ok